### PR TITLE
[@types/chrome] Add chrome.offscreen API

### PR DESCRIPTION
**Pull request by laplus-sadness**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [chrome.offscreen](https://developer.chrome.com/docs/extensions/reference/offscreen/).
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by laplus-sadness** [comment]
The tests failed because I always format the code with prettier, but didn't commit the format changes because they were over 100 lines.
After the test failed, I commited the changes. If you don't like the fact that there are many changes, please say so I can rebase this PR.

File: types/chrome/index.d.ts [hunk 1]
```diff
@@ -5998,6 +5998,63 @@ declare namespace chrome.notifications {
...
+     * Creates a new offscreen document for the extension.
+     * @param parameters The parameters describing the offscreen document to create.
+     * @param callback Invoked when the offscreen document is created and has completed its initial page load.
+     */
+    export function createDocument(parameters: CreateParameters, callback: () => void): void;
+    /**
+     * Closes the currently-open offscreen document for the extension.
+     * @param callback Invoked when the offscreen document has been closed.
+     */
+    export function closeDocument(callback: () => void): void;
```
Comment (matcarlson) [inline types/chrome/index.d.ts:6054]: Can you change this to return `Promise<void>;` so it matches other parts of this file?

File: types/chrome/index.d.ts [hunk 2]
```diff
@@ -5998,6 +5998,63 @@ declare namespace chrome.notifications {
     export function getPermissionLevel(callback: (level: string) => void): void;
 }
 
+////////////////////
+// Offscreen
+////////////////////
+/**
+ * Use the offscreen API to create and manage offscreen documents.
+ * Availability: Since Chrome 109. Manifest v3.
```
Comment (matcarlson) [inline types/chrome/index.d.ts:6006]: Can you change this to `@since Chrome 109. Manifest v3.` ?

**Review by matcarlson — CHANGES_REQUESTED** [review]
I'm cool with the whitespace changes - IMO it's easier to read.

**Comment by matcarlson** [comment]
Originally from discussion #63924

File: types/chrome/index.d.ts (hunk 1 above)
Comment (laplus-sadness) [reply types/chrome/index.d.ts:6054]: In the [documentation](https://developer.chrome.com/docs/extensions/reference/offscreen/), it's saying that, by not specifying the callback, the return will be async. However, as this overloading method is listed as _Pending_, I didn't add it.
Would you like that I list these two overloading options or just change the return to a promise?

Comment (matcarlson) [reply types/chrome/index.d.ts:6054]: Thanks for clarifying for me, I missed the pending note. I'd leave it as-is for now.

File: types/chrome/index.d.ts (hunk 2 above)
Comment (laplus-sadness) [reply types/chrome/index.d.ts]: Sorry, forgot about this.
Resolved in 2f0e8940336652e0ddd044a729ea5d171ec88dde

**Comment by laplus-sadness** [comment]
Don't review this yet @matcarlson. I sent a message to [this thread](https://github.com/GoogleChrome/chrome-extensions-samples/pull/797) after discovering that they uploaded an example usage of this api and they removed the pending label.

If someone replies to my last message, I will commit some changes over the method definitions.

**Comment by laplus-sadness** [comment]
@matcarlson I added the two different overload methods and update the tests in cc6bf5e.

Additionally, because the official example from the chrome repo uses the `hasDocument()` function and it's added to the source code, but not to the documentation, I considered that it should be added here. If they remove the function or update the example, it will need to be removed.

This PR is ready for review again.

**Review by matcarlson — APPROVED** [review]
Looks good - Thanks for your work!

**Comment by laplus-sadness** [comment]
Ready to merge

**Pull request merged**